controllers/numaresourcesoperator_controller.go

```diff
@@ -302,20 +308,26 @@ func (r *NUMAResourcesOperatorReconciler) syncDaemonSetsStatuses(ctx context.Con
 		if !isDaemonSetReady(&ds) {
 			return dssWithReadyStatus, false, nil
 		}
-		dssWithReadyStatus = append(dssWithReadyStatus, nname)
+		dssWithReadyStatus = append(dssWithReadyStatus, dsInfo.Daemonset)
 	}
 	return dssWithReadyStatus, true, nil
 }
 
-func syncNodeGroupsStatus(instance *nropv1.NUMAResourcesOperator, dsPerMCP map[string]nropv1.NamespacedName) []nropv1.NodeGroupStatus {
+func syncNodeGroupsStatus(instance *nropv1.NUMAResourcesOperator, dsPerMCP []MCPDaemonSetPair) []nropv1.NodeGroupStatus {
 	ngStatuses := []nropv1.NodeGroupStatus{}
 	for _, mcp := range instance.Status.MachineConfigPools {
-		status := nropv1.NodeGroupStatus{
-			PoolName:  mcp.Name,
-			Config:    *mcp.Config,
-			DaemonSet: dsPerMCP[mcp.Name],
+		for _, info := range dsPerMCP {
+			if mcp.Name != info.MCPName {
+				continue
+			}
+
+			status := nropv1.NodeGroupStatus{
+				PoolName:  mcp.Name,
+				Config:    *mcp.Config,
+				DaemonSet: info.Daemonset,
+			}
+			ngStatuses = append(ngStatuses, status)
```

Comment:
thanks for the fix. We can use another fix later: this function should create the `NodeGroupStatus` using `dsPerMCP` as master source (iterating over there) and not depend on other status fields like `instance.Status.MachineConfigPools`.

Reply:
we need to be aligned with the status update on MCP, currently the config won't be updated there if MCP is not completely ready. I'll look into this anyhow, we can use an improvement here indeed